--- origsrc/autogen-5.16.2/autoopts/Makefile.am	2013-10-07 19:12:41.095124600 -0500
+++ src/autogen-5.16.2/autoopts/Makefile.am	2013-10-07 19:11:55.000000000 -0500
@@ -119,7 +119,7 @@ m4datadir                 = $(datadir)/a
 nodist_libopts_la_SOURCES = libopts.c
 libopts_la_SOURCES        = $(HDRS)
 libopts_la_CFLAGS         = -DPKGDATADIR='"$(pkgdatadir)"'
-libopts_la_LDFLAGS        = -version-info $(LIBOPTS_VER)
+libopts_la_LDFLAGS        = -version-info $(LIBOPTS_VER) -no-undefined
 libopts_la_LIBADD         = $(top_builddir)/snprintfv/libsnprintfv.la
 
 EXTRA_DIST += $(SRC) $(EXTRA_DATA) $(man_MANS) $(DEF_FILES)
