--- origsrc/autogen-5.9.7/autoopts/Makefile.am	2008-11-16 17:56:59.000000000 -0600
+++ src/autogen-5.9.7/autoopts/Makefile.am	2009-04-21 23:26:19.233750000 -0500
@@ -102,13 +102,13 @@ m4datadir                 = $(datadir)/a
 nodist_libopts_la_SOURCES = libopts.c
 libopts_la_SOURCES        = $(HDRS)
 libopts_la_CFLAGS         = -DPKGDATADIR='"$(pkgdatadir)"'
-libopts_la_LDFLAGS        = -version-info $(LIBOPTS_VER)
+libopts_la_LDFLAGS        = -version-info $(LIBOPTS_VER) -no-undefined
 libopts_la_LIBADD         = $(top_builddir)/snprintfv/libsnprintfv.la
 
 libguileopts_la_SOURCES   = guileopt.c
 libguileopts_la_CFLAGS    = $(LIBGUILE_CFLAGS)
-libguileopts_la_LIBADD    = libopts.la
-libguileopts_la_LDFLAGS   = -version-info $(LGUILE_VER)
+libguileopts_la_LIBADD    = libopts.la $(LIBGUILE_LIBS)
+libguileopts_la_LDFLAGS   = -version-info $(LGUILE_VER) -no-undefined
 
 EXTRA_DIST = $(SRC) $(pkgdata_DATA) $(man_MANS) \
 	$(MISC_FILES) $(DEF_FILES) install-options-h mk-autoopts-pc.in po
