--- origsrc/autogen-5.9.3/autoopts/Makefile.am	2007-09-08 11:23:26.000000000 -0500
+++ src/autogen-5.9.3/autoopts/Makefile.am	2007-11-25 21:24:04.281250000 -0600
@@ -95,12 +95,12 @@
 nodist_libopts_la_SOURCES = libopts.c
 libopts_la_SOURCES        = $(HDRS)
 libopts_la_CFLAGS         = -DPKGDATADIR='"$(pkgdatadir)"'
-libopts_la_LDFLAGS        = -version-info $(LIBOPTS_VER)
+libopts_la_LDFLAGS        = -version-info $(LIBOPTS_VER) -no-undefined
 
 libguileopts_la_SOURCES   = guileopt.c
 libguileopts_la_CFLAGS    = $(LIBGUILE_CFLAGS)
-libguileopts_la_LIBADD    = libopts.la
-libguileopts_la_LDFLAGS   = -version-info $(LGUILE_VER)
+libguileopts_la_LIBADD    = libopts.la $(LIBGUILE_LIBS)
+libguileopts_la_LDFLAGS   = -version-info $(LGUILE_VER) -no-undefined
 
 EXTRA_DIST = $(HDRS) $(CSRC) $(pkgdata_DATA) $(man_MANS) $(TPL_FILES) \
 	$(MISC_FILES) $(DEF_FILES) install-options-h mk-autoopts-pc.in po
